solutions/swb-reference/integration-tests/support/complex/accountHelper.ts

```diff
@@ -0,0 +1,153 @@
+    const busRule = await this._awsSdk.clients.eventBridge.describeRule(describeRuleParams);
+
+    const putRuleParams = {
+      Name: busRuleName,
+      EventPattern: JSON.stringify({
+        account: _.difference(JSON.parse(busRule.EventPattern!).account, [awsAccountId]),
+        source: [{ 'anything-but': ['aws.config', 'aws.cloudtrail', 'aws.ssm', 'aws.tag'] }],
+        'detail-type': [{ 'anything-but': 'AWS API Call via CloudTrail' }]
+      }),
+      EventBusName: busName
+    };
+
+    // Update rule for main account event bus
+    await this._awsSdk.clients.eventBridge.putRule(putRuleParams);
+  }
+
+  public async updateMainAccountEncryptionKeyPolicy(
+    awsAccountId: string
+  ): Promise<void> {
+    const mainAcctEncryptionArn = this._settings.get('MainAccountEncryptionKeyOutput');
+    const keyId = mainAcctEncryptionArn.split('/').pop()!;
+    const keyPolicyResponse = await this._awsSdk.clients.kms.getKeyPolicy({
+      KeyId: keyId,
+      PolicyName: 'default'
+    });
+    let keyPolicy = PolicyDocument.fromJson(JSON.parse(keyPolicyResponse.Policy!));
+
+    keyPolicy = IamHelper.removePrincipalFromStatement(
+      keyPolicy,
+      'main-key-share-statement',
+      `arn:aws:iam::${awsAccountId}:root`
+    );
+
+    const putPolicyParams = {
+      KeyId: keyId,
+      PolicyName: 'default',
+      Policy: JSON.stringify(keyPolicy.toJSON())
+    };
+
+    // Update key policy
+    await this._awsSdk.clients.kms.putKeyPolicy(putPolicyParams);
+  }
+
+  public async updateArtifactsBucketPolicy(awsAccountId: string): Promise<void> {
+    const bucketName = this._settings.get('S3BucketArtifactsArnOutput').split(':').pop();
+    const bucketPolicyResponse: GetBucketPolicyCommandOutput = await this._awsSdk.clients.s3.getBucketPolicy({
+      Bucket: bucketName
+    });
+    let bucketPolicy;
+    bucketPolicy = PolicyDocument.fromJson(JSON.parse(bucketPolicyResponse.Policy!));
+
+    bucketPolicy = IamHelper.removePrincipalFromStatement(
+      bucketPolicy,
+      'List:environment-files',
+      `arn:aws:iam::${awsAccountId}:root`
+    );
+
+    bucketPolicy = IamHelper.removePrincipalFromStatement(
+      bucketPolicy,
+      'Get:environment-files',
+      `arn:aws:iam::${awsAccountId}:root`
+    );
+
+    const putPolicyParams: PutBucketPolicyCommandInput = {
+      Bucket: bucketName,
+      Policy: JSON.stringify(bucketPolicy.toJSON())
+    };
+
+    // Update bucket policy
+    await this._awsSdk.clients.s3.putBucketPolicy(putPolicyParams);
+  }
+
+  public async deOnboardAccount(awsAccountId: string): Promise<void> {
+  // Undo all operations that happen in: hostingAccountLifecycleService.initializeAccount()
+
+  // Update main account default event bus to remove hosting account state change events
+  await this.updateBusPermissions(awsAccountId);
+
+  // Remove account to artifactBucket's bucket policy
+  await this.updateArtifactsBucketPolicy(awsAccountId);
```

Comment:
Let's incorporate part of the comment in line 147 into `updateArtifactsBucketPolicy` name. Maybe `removeAccountFromArtifactBucketPolicy`